File: codes/go/route/astronaut.go
package route

import (
  "net/http"
  "strconv"

	"github.com/gin-gonic/gin"

  "github.com/nicolas-grevin/green-it/repository"
  "github.com/nicolas-grevin/green-it/service"
)

type astronautRoute struct {
  repo *repository.AstronautRepository
}

func AstronautRouter(svc *service.PostgresService, r *gin.RouterGroup) {
  ar := astronautRoute{
    repo: repository.NewAstronautRepository(svc.DB),
  }

  agr := r.Group("/astronauts")
  agr.GET("", ar.list)
  agr.GET("/:id", ar.view)
}

func (ar astronautRoute) list(c *gin.Context) {
  astronauts, err := ar.repo.List()

  if err != nil {
    c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})

    return
  }
  
  c.JSON(http.StatusOK, astronauts)
} 

func (ar astronautRoute) view(c *gin.Context) {
  id, err := strconv.ParseUint(c.Param("id"), 10, 64)

  if err != nil {
    panic(err)
  }

  astronaut, err := ar.repo.FindById(uint16(id))

	if err != nil {
		c.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
	
    return
  }

	c.JSON(http.StatusOK, astronaut)
}

File: codes/deno/route/astronaut.ts
import { Hono, Context } from "hono";
import { Client } from "postgres";

import { RouteHandler } from "./route.ts";

import { AstronautRepository } from "../repository/astronaut.ts";

export class AstronautHandler implements RouteHandler {
  private _repository: AstronautRepository;
  private _router: Hono

  constructor(client: Client) {
    this._repository = new AstronautRepository(client);
    this._router = new Hono();
  }

  getRoutes(): Hono {
    this._router.get("", (c: Context) => this.list(c))
    this._router.get("/:id", (c: Context) => this.view(c))

    return this._router
  }

  private async list(c: Context) {
    try {
      return c.json(await this._repository.list(), 200);
    } catch (e) {
      return c.json({"error": e}, 200);
    }
  }

  private async view(c: Context) {
    try {
      const { id } = c.req.param();

      return c.json(await this._repository.findById(id), 200);
    } catch (e) {
      return c.json({"error": e}, 200);
    }
  }
}
